src/main/java/net/mcreator/ui/workspace/WorkspacePanel.java

```diff
@@ -857,6 +857,14 @@ public void switchFolder(FolderElement switchTo) {
 
 		upFolder.setEnabled(!currentFolder.isRoot());
 	}
+	
+	private void togglefilter(String filter) {
+		if (!Pattern.matches(".*" + filter + ".*", search.getText())) {
+			search.setText(search.getText() + " " + filter);
+		} else {
+			search.setText(search.getText().replaceAll(" " + filter, "").replaceAll(" ", " "));

```

Comment:
`.replaceAll(" ", " ")` this does not reduce multiple spaces into one

Reply:
is this part really nessesary though? i mean, i did my own testings with possible combination and i don't really find it nessassary as having 2 space doesn't happen

Reply:
Yes it is, and don't resolve convos yourself a this was by no means resolved. Also there was no need for Pattern matches here and .* regex. I really recommend learning Java well ;)